pkg/processor/runtime/python/runtime.go

```diff
@@ -264,6 +268,19 @@ func (py *python) handleResponseLog(functionLogger nuclio.Logger, response []byt
 	logFunc(message, vars...)
 }
 
+func (py *python) handleReponseMetric(functionLogger nuclio.Logger, response []byte) {
+
+	metrics := make(map[string]interface{})
+
+	if err := json.Unmarshal(response, &metrics); err != nil {
+		functionLogger.ErrorWith("Can't decode metric", "error", err)
+		return
+	}
+
+	// TODO: Push metrics?
```

Comment:
Concept of metrics is that handing prometheus metric objects and moreover pushing is decoupled from runtimes for performance reasons. They just need to update the lock free `Statistics` structure which is part of the `AbstractRuntime`. A generic mechanism does gathering. Since these are natural words (uint64) the read/write contention is negated by the underlying architecture and we are therefore lock free.

Long story short - you need to update `py.Statistics` here (also please remove the log).

Reply:
OK